File: lib/models/auto_scheduled_post.dart
class AutoScheduledPost {
  final int id;
  final String userId;
  final String content;
  final List<String> mediaUrls;
  final List<String> platforms;
  final DateTime scheduleTime;
  final String recurrencePattern;
  final int? recurrenceInterval;
  final DateTime? recurrenceEndDate;
  final bool isActive;
  final String status;
  final DateTime? lastPostedAt;
  final DateTime? nextPostAt;
  final int postCount;

  AutoScheduledPost({
    required this.id,
    required this.userId,
    required this.content,
    required this.mediaUrls,
    required this.platforms,
    required this.scheduleTime,
    required this.recurrencePattern,
    this.recurrenceInterval,
    this.recurrenceEndDate,
    required this.isActive,
    required this.status,
    this.lastPostedAt,
    this.nextPostAt,
    required this.postCount,
  });

  factory AutoScheduledPost.fromJson(Map<String, dynamic> json) {
    return AutoScheduledPost(
      id: json['id'] as int,
      userId: json['user_id'] as String,
      content: json['content'] as String,
      mediaUrls: json['media_urls'] != null
          ? List<String>.from(json['media_urls'] as List)
          : [],
      platforms: json['platforms'] != null
          ? List<String>.from(json['platforms'] as List)
          : [],
      scheduleTime: DateTime.parse(json['schedule_time'] as String),
      recurrencePattern: json['recurrence_pattern'] as String,
      recurrenceInterval: json['recurrence_interval'] as int?,
      recurrenceEndDate: json['recurrence_end_date'] != null
          ? DateTime.parse(json['recurrence_end_date'] as String)
          : null,
      isActive: json['is_active'] as bool? ?? false,
      status: json['status'] as String? ?? 'pending',
      lastPostedAt: json['last_posted_at'] != null
          ? DateTime.parse(json['last_posted_at'] as String)
          : null,
      nextPostAt: json['next_post_at'] != null
          ? DateTime.parse(json['next_post_at'] as String)
          : null,
      postCount: json['post_count'] as int? ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'user_id': userId,
      'content': content,
      'media_urls': mediaUrls,
      'platforms': platforms,
      'schedule_time': scheduleTime.toIso8601String(),
      'recurrence_pattern': recurrencePattern,
      'recurrence_interval': recurrenceInterval,
      'recurrence_end_date': recurrenceEndDate?.toIso8601String(),
      'is_active': isActive,
    };
  }

  bool get isPending => status == 'pending';
  bool get isStatusActive => status == 'active';  // Changed from isActive to avoid conflict
  bool get isPaused => status == 'paused';
  bool get isCompleted => status == 'completed';
  bool get isFailed => status == 'failed';

  String get statusText {
    switch (status) {
      case 'pending':
        return 'قيد الانتظار';
      case 'active':
        return 'نشط';
      case 'paused':
        return 'متوقف مؤقتاً';
      case 'completed':
        return 'مكتمل';
      case 'failed':
        return 'فشل';
      default:
        return status;
    }
  }

  String get recurrencePatternText {
    switch (recurrencePattern) {
      case 'once':
        return 'مرة واحدة';
      case 'daily':
        return 'يومي';
      case 'weekly':
        return 'أسبوعي';
      case 'monthly':
        return 'شهري';
      case 'custom':
        return 'مخصص';
      default:
        return recurrencePattern;
    }
  }

  String get platformsText {
    if (platforms.isEmpty) return 'لا توجد منصات';
    return platforms.join(', ');
  }
}

File: backend/app/Models/AutoScheduledPost.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class AutoScheduledPost extends Model
{
    protected $fillable = [
        'user_id',
        'content',
        'media_urls',
        'platforms',
        'schedule_time',
        'recurrence_pattern',
        'recurrence_interval',
        'recurrence_end_date',
        'is_active',
        'status',
        'last_posted_at',
        'next_post_at',
        'post_count',
        'metadata',
    ];

    protected $casts = [
        'media_urls' => 'array',
        'platforms' => 'array',
        'schedule_time' => 'datetime',
        'recurrence_end_date' => 'datetime',
        'last_posted_at' => 'datetime',
        'next_post_at' => 'datetime',
        'is_active' => 'boolean',
        'metadata' => 'array',
        'post_count' => 'integer',
    ];

    // Status constants
    const STATUS_PENDING = 'pending';
    const STATUS_ACTIVE = 'active';
    const STATUS_PAUSED = 'paused';
    const STATUS_COMPLETED = 'completed';
    const STATUS_FAILED = 'failed';

    // Recurrence pattern constants
    const PATTERN_ONCE = 'once';
    const PATTERN_DAILY = 'daily';
    const PATTERN_WEEKLY = 'weekly';
    const PATTERN_MONTHLY = 'monthly';
    const PATTERN_CUSTOM = 'custom';

    // Relationships
    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class, 'user_id', 'id');
    }

    // Accessors
    public function getStatusTextAttribute(): string
    {
        return match($this->status) {
            self::STATUS_PENDING => 'قيد الانتظار',
            self::STATUS_ACTIVE => 'نشط',
            self::STATUS_PAUSED => 'متوقف مؤقتاً',
            self::STATUS_COMPLETED => 'مكتمل',
            self::STATUS_FAILED => 'فشل',
            default => $this->status,
        };
    }

    public function getRecurrencePatternTextAttribute(): string
    {
        return match($this->recurrence_pattern) {
            self::PATTERN_ONCE => 'مرة واحدة',
            self::PATTERN_DAILY => 'يومي',
            self::PATTERN_WEEKLY => 'أسبوعي',
            self::PATTERN_MONTHLY => 'شهري',
            self::PATTERN_CUSTOM => 'مخصص',
            default => $this->recurrence_pattern,
        };
    }

    // Methods
    public function isPending(): bool
    {
        return $this->status === self::STATUS_PENDING;
    }

    public function isActive(): bool
    {
        return $this->status === self::STATUS_ACTIVE;
    }

    public function isPaused(): bool
    {
        return $this->status === self::STATUS_PAUSED;
    }

    public function isCompleted(): bool
    {
        return $this->status === self::STATUS_COMPLETED;
    }

    public function activate(): bool
    {
        $this->status = self::STATUS_ACTIVE;
        $this->is_active = true;
        return $this->save();
    }

    public function pause(): bool
    {
        $this->status = self::STATUS_PAUSED;
        $this->is_active = false;
        return $this->save();
    }

    public function complete(): bool
    {
        $this->status = self::STATUS_COMPLETED;
        $this->is_active = false;
        return $this->save();
    }

    public function calculateNextPostTime(): void
    {
        if (!$this->last_posted_at) {
            $this->next_post_at = $this->schedule_time;
            return;
        }

        switch ($this->recurrence_pattern) {
            case self::PATTERN_ONCE:
                $this->next_post_at = null;
                $this->complete();
                break;

            case self::PATTERN_DAILY:
                $this->next_post_at = $this->last_posted_at->addDays($this->recurrence_interval ?? 1);
                break;

            case self::PATTERN_WEEKLY:
                $this->next_post_at = $this->last_posted_at->addWeeks($this->recurrence_interval ?? 1);
                break;

            case self::PATTERN_MONTHLY:
                $this->next_post_at = $this->last_posted_at->addMonths($this->recurrence_interval ?? 1);
                break;

            case self::PATTERN_CUSTOM:
                if ($this->recurrence_interval) {
                    $this->next_post_at = $this->last_posted_at->addHours($this->recurrence_interval);
                }
                break;
        }

        // Check if we've passed the end date
        if ($this->recurrence_end_date && $this->next_post_at > $this->recurrence_end_date) {
            $this->complete();
        }

        $this->save();
    }

    public function markAsPosted(): void
    {
        $this->last_posted_at = now();
        $this->post_count = ($this->post_count ?? 0) + 1;
        $this->calculateNextPostTime();
        $this->save();
    }

    public function shouldPost(): bool
    {
        if (!$this->is_active || !$this->isActive()) {
            return false;
        }

        if (!$this->next_post_at) {
            return false;
        }

        return $this->next_post_at <= now();
    }

    // Scopes
    public function scopeActive($query)
    {
        return $query->where('is_active', true)->where('status', self::STATUS_ACTIVE);
    }

    public function scopePending($query)
    {
        return $query->where('status', self::STATUS_PENDING);
    }

    public function scopePaused($query)
    {
        return $query->where('status', self::STATUS_PAUSED);
    }

    public function scopeDueForPosting($query)
    {
        return $query->active()
            ->whereNotNull('next_post_at')
            ->where('next_post_at', '<=', now());
    }

    public function scopeRecent($query)
    {
        return $query->orderBy('created_at', 'desc');
    }
}
